src/mscorlib/src/System/Diagnostics/Eventing/EventSource.cs

```diff
@@ -250,6 +254,10 @@ namespace System.Diagnostics.Tracing
     /// </remarks>
     public partial class EventSource : IDisposable
     {
+#if FEATURE_CORECLR && FEATURE_PAL   
```

Comment:
Instead of having FEATURE_CORECLR && FEATURE_PAL, I would suggest that we have something like FEATURE_EVENTSOURCE_XPLAT - that is more readable and provides context about the nature of the feature.

Using FEATURE_PAL in the VM/FX is an ugly artifact of the past that, semantically, is not applicable in all scenarios (this being one of them) since this is not part of the PAL.